src/_tutorials/dart-vm/httpserver.md

```diff
@@ -555,66 +560,55 @@ The `close()` method sends the request to the server
 and returns the second Future, which completes with
 an HttpClientResponse object.
 
+<?code-excerpt "httpserver/bin/basic_writer_client.dart" replace="/\/\*.\*\/|UTF8.\w+/[!$&!]/g"?>
 {% prettify dart %}
-   import 'dart:io';
-   import 'dart:convert' show UTF8, JSON;
-
-   main() async {
-     Map jsonData = {
-       'name':     'Han Solo',
-       'job':      'reluctant hero',
-       'BFF':      'Chewbacca',
-       'ship':     'Millennium Falcon',
-       'weakness': 'smuggling debts'
-     };
-
-[[note]]1[[/note]]  [[highlight]]var request = await[[/highlight]] new HttpClient().post(
-[[note]]2[[/note]]      [[highlight]]InternetAddress.LOOPBACK_IP_V4.host, 4049, '/file.txt');[[/highlight]]
-[[note]]3[[/note]]  [[highlight]]request.headers.contentType = ContentType.JSON;[[/highlight]]
-[[note]]4[[/note]]  [[highlight]]request.write(JSON.encode(jsonData));[[/highlight]]
-[[note]]5[[/note]]  [[highlight]]HttpClientResponse response = await request.close()[[/highlight]];
-[[note]]6[[/note]]  await for (var contents in response.transform([[highlight]]UTF8.decoder[[/highlight]])) {
-       print(contents);
-    }
+import 'dart:async';
+import 'dart:io';
+import 'dart:convert' show UTF8, JSON;
+
+Future main() async {
+  Map jsonData = {
+    'name': 'Han Solo',
+    'job': 'reluctant hero',
+    'BFF': 'Chewbacca',
+    'ship': 'Millennium Falcon',
+    'weakness': 'smuggling debts'
+  };
+
+  var request = await new HttpClient()
+      [!/*1*/!] .post(InternetAddress.LOOPBACK_IP_V4.host, 4049, 'file.txt');
```

Comment:
Yes, in part because of line issues, but also because dartfmt doesn't make it easy to place the first marker close to the `post(` tokens. I've moved the markers. Let me know what you think about their new positions.